service/history/shard/fx.go

```diff
@@ -25,16 +25,25 @@
 package shard
 
 import (
+	"go.temporal.io/server/service/history/configs"
 	"go.uber.org/fx"
 
 	"go.temporal.io/server/common"
 )
 
-var Module = fx.Options(
-	fx.Provide(ControllerProvider),
-	fx.Provide(ContextFactoryProvider),
-	fx.Provide(fx.Annotate(
```

Comment:
Two reasons for removing this:
1. fx.Provide is variadic, so you don't need multiple statements
2. fx.Options usage is discouraged; the docs read, "Use this pattern sparingly, since it limits the user's ability to customize their application." On its own, it's fine, but it allows you to embed other modules, which induces the problem that quote is talking about because you're choosing the inputs to the module for the user instead of letting them do so.

Reply:
Does fx.Provide work with Decorate or Replace? I remember using fx.Module can limit the scope of those functions, which in most cases are ideal. I kinda think we should change all `var Module = fx.Options(...)` in our codebase to a real fx.Module.

Reply:
Decorate and Replace can be used by clients of this module just as easily as they could if this were an `fx.Module`. I don't think there's a point in libraries defining their own `fx.Module` objects because, if a client wants to limit the scope of a graph augmentor (decorate or replace), their code will basically look the exact same: `fx.Module(name, library.SomeModule, fx.{Decorate,Replace}(...))` regardless of whether the library `SomeModule` is defined using `fx.Provide`, `fx.Options` or `fx.Module` because the clients still need to define their own module around it in order to limit their augmentor's scope--they can't put anything inside of an already-defined module.